### DISCUSSION TITLE
Are you not able to use DG kernels and MOOSE FV objects?

### DISCUSSION THREAD

**Discussion by aladshaw3** [idaholab/moose#19128] [Q&A Modules: Navier-Stokes] [answered]
Hello,

I am doing some testing with the incompressible Navier-Stokes module with MOOSE FVs. I wanted to couple the velocity field from FVs with another set of kernels for mass transport using DG methods. However, if I activate any DG kernels I get the following.

*** ERROR ***
dofIndicesLower not supported by MooseVariableFVBase

I also get this even if no FV variable is coupled to any non-FV variable. Is it not possible to have DG kernels active with FV stuff in MOOSE?

**Comment by GiudGiud** [comment]
Hello

There's very limited support for FE-FV coupling right now. We have this PR that will start to improve this: https://github.com/idaholab/moose/pull/18394
but I dont know that it includes DG kernels. 

What are you trying to do with those DG kernels? It s pretty likely it can be done in FV already, and if not we may want to add it

Guillaume

**Comment by aladshaw3** [comment, marked as answer]
@GiudGiud 
Several years back, I had started developing a series of DG kernels for use in my own MOOSE modules for catalysis and adsorption (https://github.com/aladshaw3/cats). I used DG kernels for improved stability and conservation laws. When I developed them, I made them such that any velocity terms could be variables and I planned to later link those kernels to a flow field simulation. I had some success coupling with the CGFE incompressible Navier-Stokes modules, but wanted to try using the FV version of those kernels for stability. 

When I had started this work, there was no FV kernels. Now that there are FV kernels, I could rework some things to do everything with FV, but that is an undertaking that may take a lot of time, so I wanted to see about coupling DG kernels and variables (which I already have coded) with the new FV system for flow fields.

**Reply by GiudGiud** [reply]
That's really cool. I hope FV improves the stability, it's easier to converge as far as I've noticed

@csdechant does your PR enable DG Kernels?

**Reply by GiudGiud** [reply]
@lindsayad @rwcarlsen since you're reviewing that PR

**Reply by lindsayad** [reply]
#18394 should be agnostic of the backing `FEType`

**Reply by lindsayad** [reply]
But @csdechant has been hard to pin down recently :smile:
